app/models/subscriber.rb

```diff
@@ -0,0 +1,32 @@
+class Subscriber
+  def self.for_discussion(discussion)
+    new(discussion)
+  end
+
+  def initialize(discussion)
+    @discussion = discussion
+    @subscriptions = Subscription.where(discussion_id: discussion.id)
+  end
+
+  def subscribe(users)
+    users.each do |user|
+      create_subscription_for_user(user)
+    end
+  end
+
+  private
+
+  attr_reader :discussion, :subscriptions
+
+  def create_subscription_for_user(user)
+    return if user_already_subscribed?(user)
+
+    Subscription.create(user: user, discussion: discussion)
+  end
+
+  def user_already_subscribed?(user)
+    subscriptions.any? do |subscription|
+      subscription.user_id == user.id
+    end
+  end
```

Comment:
Ok I see that now

Just as a suggestion, if your goal is to optimize number of queries, there's a couple other things you could look at:

* filter all users in a single query, instead of using a guard to check them individually. i.e. `users.where.not(id: subscriptions.pluck(:user_id))`
* create all subscriptions in a single query, eliminating another N+1 problem: `Subscription.create([{user: user1, discussion: discussion}, {...}])`

but these are just suggestions. they might even not be doable without making the code much less readable